tensorflow/cc/gradients/math_grad.cc

```diff
@@ -687,6 +687,22 @@ Status MeanGrad(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("Mean", MeanGrad);
 
+Status ErfGrad(const Scope& scope, const Operation& op,
+               const std::vector<Output>& grad_inputs,
+               std::vector<Output>* grad_outputs) {
+  auto grad = grad_inputs[0];
+  auto two_over_root_pi = Cast(scope, Const(scope, 2 / std::sqrt(M_PI)),
+                               grad.type());
+  Scope grad_scope = scope.WithControlDependencies(grad);
+  auto x = ConjugateHelper(grad_scope, op.input(0));
+  auto dx = Mul(scope,
```

Comment:
Did you intend to use grad_scope for these scopes below?

Reply:
Hi, @suharshs . 

`grade_scope` is used to keep consistent with its corresponding implementation in python.

```python
def _ErfGrad(op, grad):
  """Returns grad * 2/sqrt(pi) * exp(-x**2)."""
  x = op.inputs[0]
  two_over_root_pi = constant_op.constant(2 / np.sqrt(np.pi), dtype=grad.dtype)
  with ops.control_dependencies([grad]):
    x = math_ops.conj(x)
    return grad * two_over_root_pi * math_ops.exp(-math_ops.square(x))
```

However, in my opinion, `with ops.control_dependencies([grad])` seems useless here. Moreover, I found a lot of gradients remove the dependency as well, take AtanhGrad for example

```python
@ops.RegisterGradient("Atanh")
def _AtanhGrad(op, grad):
  """Returns grad * 1/ (1 - x^2)."""
  x = op.inputs[0]
  with ops.control_dependencies([grad]):
    x = math_ops.conj(x)
    x2 = math_ops.square(x)
    one = constant_op.constant(1, dtype=grad.dtype)
    inv = math_ops.reciprocal(math_ops.subtract(one, x2))
    return grad * inv
```

```c++
Status AtanhGrad(const Scope& scope, const Operation& op,
                 const std::vector<Output>& grad_inputs,
                 std::vector<Output>* grad_outputs) {
  // y = atanh(x)
  // dy/dx = 1 / (1 - x^2)
  auto one = Cast(scope, Const(scope, 1.0), op.input(0).type());
  auto dydx = Reciprocal(scope, Sub(scope, one, Square(scope, op.input(0))));
  // grad(x) = grad(y) * conj(dy/dx)
  grad_outputs->push_back(
      Mul(scope, grad_inputs[0], ConjugateHelper(scope, dydx)));
  return scope.status();
}
```

Above all, I prefer to eliminating the useless dependency. How do you think? Thanks.

Reply:
By the way, to write the PR, I refer to [SigmoidGrad](https://github.com/tensorflow/tensorflow/blob/1f19b8cf38dcf7b468cb7f4e6d969966a495e167/tensorflow/cc/gradients/math_grad.cc#L259) which is introduced by commit 8624ecc9e827a40f9b514ff6b8ed925390ca79cc
```c++
auto y = ConjugateHelper(grad_scope, op.output(0));
grad_outputs->push_back(internal::SigmoidGrad(scope, y, grad));
```

To be honest, I agree that `grad_scope` should be used below, rather than `scope`.  But the author of the commit seems a robot, right?